tests/h/services/search_index/_queue_test.py

```diff
@@ -0,0 +1,364 @@
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Syncing 10 annotations that are different in Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing(annotation_ids[:limit]).only()
+        )
+
+    def test_if_there_are_multiple_jobs_with_the_same_annotation_id(
+        self,
+        annotation_ids,
+        batch_indexer,
+        caplog,
+        es,
+        queue,
+    ):
+        queue.add_all(
+            [annotation_ids[0], annotation_ids[0], annotation_ids[0]],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        # It only retrieves the annotation from Elasticsearch once.
+        es.conn.search.assert_called_once_with(
+            body=Any.dict.containing(
+                {
+                    "query": Any.dict.containing(
+                        {
+                            "ids": {
+                                "values": [annotation_ids[0]],
+                            },
+                        },
+                    ),
+                },
+            ),
+        )
+
+        # It only syncs the annotation to Elasticsearch once.
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Syncing 1 annotations that are missing from Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing([annotation_ids[0]]).only()
+        )
+
+    def test_deleting_multiple_jobs_with_the_same_annotation_id(
+        self,
+        annotations,
+        batch_indexer,
+        caplog,
+        db_session,
+        es,
+        queue,
+    ):
+        queue.add_all(
+            [annotations[0].id, annotations[0].id, annotations[0].id],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+        es.conn.search.return_value = {
+            "hits": {
+                "hits": [
+                    {
+                        "_id": annotations[0].id,
+                        "_source": {
+                            "updated": (annotations[0].updated).isoformat(),
+                        },
+                    }
+                ],
+            },
+        }
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Deleting 3 successfully completed jobs from the queue",
+            )
+        ]
+        assert db_session.query(Job).all() == []
+        batch_indexer.index.assert_not_called()
+
+
+one_week_from_now = datetime.datetime.utcnow() + datetime.timedelta(weeks=1)
+
+
+five_minutes_ago = datetime.datetime.utcnow() - datetime.timedelta(minutes=5)
+
+
+limit = 10
```

Comment:
```suggestion
LIMIT = 10
ONE_WEEK_FROM_NOW = datetime.datetime.utcnow() + datetime.timedelta(weeks=1)
FIVE_MINUTES_AGO = datetime.datetime.utcnow() - datetime.timedelta(minutes=5)
```

These are super confusing. With fixtures and such there are a lot of interestingly scoped things going about. Upper casing them suggests to the reader they are just constants. As such I'd also stick them at the top of the file.

Reply:
Yeah, these should be capitalised, done.

Let's keep them at the bottom of the file though -- our style is that constants, helper functions, fixtures etc go at the bottom of the file or class whenever possible. The thinking is that the actual tests (or source code for source files) is the interesting thing that I want to get to first, I don't want to scroll past a bunch of constants or all the private helper functions etc before getting to the actual code. Those can be hidden away at the bottom where you can go and look at them if you need to know what they are, but otherwise they don't get in your way. It also keeps the constants together with the fixtures and helpers which're already at the bottom.